examples/advanced_agents/sandbox_agent/src/nat_sandbox_agent/sandbox/docker_sandbox.py

```
@@ -0,0 +1,306 @@
+                if file_obj:
+                    return file_obj.read().decode("utf-8", errors="replace")
+                raise FileNotFoundError(f"File not found: {path}")
+
+        try:
+            # Wrap all blocking I/O in run_in_executor
+            return await asyncio.get_running_loop().run_in_executor(
+                None, _extract_from_archive
+            )
+
+        except NotFound:
+            raise FileNotFoundError(f"File not found: {path}")
```

Ruff (0.14.13)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)